### PR TITLE
Convert to python module (pyproject.toml)

### PR DESCRIPTION
**Pull request by VannTen**
possible followup to #658

### CODE REVIEW

File: producer.py [hunk 1]
```diff
@@ -46,14 +43,23 @@
 from thoth.python import AIOSource
 from thoth.python.exceptions import NotFoundError
 
+try:
```
Comment (harshad16) [inline producer.py:46]: The reason the `__version__` was kept in the version.py 
is as we use kebechet for release, it currently only recognizes these list of files.
https://github.com/thoth-station/kebechet/blob/4c52e72a0ee0b9299b5165d77cc50c33b58f0350/kebechet/managers/version/release_triggers.py#L104

**Review by harshad16 — COMMENTED** [review]
lgtm
just one thought , i would like to share.

File: producer.py (hunk 1 above)
Comment (VannTen) [reply producer.py:46]: Hum.
Relevant:
- https://github.com/thoth-station/kebechet/issues/1062 (probably)
-  https://github.com/thoth-station/core/issues/360

Do you mean kebechet require handling version in those way to handle releases ? Should'nt that be separate ?
(I don't see support for pyproject.toml, setup.cfg / others here. Would that mean that python packages using those can't use kebechet for that purpose ?)

Comment (harshad16) [reply producer.py:46]: yes, correct, there isn't support for these in kebechet 
thanks for the issue, let's make kebechet more better. :+1:

File: producer.py [hunk 2]
```diff
@@ -46,14 +43,23 @@
...
+    __version__ = version(__package__)
+except PackageNotFoundError:
+    __version__ = "v0.11.10"  # TODO makes this unnecessary by fixing users of the package
```
Comment (KPostOffice) [inline producer.py:49]: Have this default to `dev`? I think local development is the most likely cause of version not being found.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+license = {file = "LICENSE"}
+dynamic = ["version"]
+
+[project.scripts]
+package-releases-job = "producer:main"
+
+[build-system]
+requires = ["setuptools", "wheel", "setuptools_scm"]
+
+[tool.setuptools_scm]
```
Comment (harshad16) [inline pyproject.toml:15]: This might not be needed for this component, as this is not released  as module, just as a container image on release.

**Review by harshad16 — CHANGES_REQUESTED** [review]
On reviewing, 
its seems, i missed earlier that this component is not module.
we don't need the project in pyproject.toml 
and can keep the version.py for this component.
as it would be released as container image.

**Comment by VannTen** [comment]
On Wed, Jan 18, 2023 at 02:26:34PM -0800, Harshad Reddy Nalla wrote:
> @harshad16 requested changes on this pull request.
> 
> On reviewing, 
> its seems, i missed earlier that this component is not module.
> we don't need the project in pyproject.toml 
> and can keep the version.py for this component.
> as it would be released as container image. 
> 
> > +[project]
> +name = "package-releases-job"
> +description = "Check packages updates on Python package indexes"
> +readme = "README.rst"
> +requires-python = ">=3.8"
> +license = {file = "LICENSE"}
> +dynamic = ["version"]
> +
> +[project.scripts]
> +package-releases-job = "producer:main"
> +
> +[build-system]
> +requires = ["setuptools", "wheel", "setuptools_scm"]
> +
> +[tool.setuptools_scm]
> 
> This might not be needed for this component, as this is not released  as module, just as a container image on release.


I don't disagree ; I still think we could get some benefits from converting to pyproject.toml, mainly standardizing all of thoth repos on a common model so we can treat them the same in the CI.

**Comment by harshad16** [comment]
> I don't disagree ; I still think we could get some benefits from converting to pyproject.toml, mainly standardizing all of thoth repos on a common model so we can treat them the same in the CI.

we are using pyproject.toml.
is there a PEP standard for having a version on pyproject.toml, it seems project which use poetry use that?
as this project is just a component, it feel version.py file is more easy and convenient. 
WDYT ?

**Comment by VannTen** [comment]
Actually we can have both if having version defined inside a file is desirable

setuptools support something like this in pyproject.toml

[project]
dynamic = ["version"]

[tool.setuptools.dynamic]
version = {attr: "version.version"}

The main point is to avoid repeating the version twice and then
forgetting to update it twice later on.

Would that work ?

**Comment by harshad16** [comment]
> Actually we can have both if having version defined inside a file is desirable setuptools support something like this in pyproject.toml [project] dynamic = ["version"] [tool.setuptools.dynamic] version = {attr: "version.version"} The main point is to avoid repeating the version twice and then forgetting to update it twice later on. Would that work ?

yes, we can do this.
side question: what is the reason to have version in pyproject.toml in projects which are not python modules ?

**Comment by VannTen** [comment]
Well, handling them as python modules allows to handle them in a more
general way. For example, using the repo as a pre-commit hook (like
we're currently doing in adviser) rely on it being a python module.

Some of the tooling (mypy in particular) seems to be a bit lost when it
can't identify a module.

Regarding scripts / job specifically, relying on [project.scripts]
instead of manually invoking the python file guarantee we'll have the
correct dependency in scope by just doing a `pip install .` of the repo.

All of the above is my opinion but TL;DR: easier integration with the
Python ecosystem and it's tooling.

**Comment by VannTen** [comment]
Do you think we can have the `__version__` directly in `producer.py` ?
Having one single file for that seems a bit strange.

(Or maybe that's related to kebechet ?)

**Comment by harshad16** [comment]
> Do you think we can have the `__version__` directly in `producer.py` ? Having one single file for that seems a bit strange. (Or maybe that's related to kebechet ?)

yeah that is related to kebechet 
maybe can plan this future

**Pull request closed without merging**